core/arch/arm/kernel/user_ta.c

```diff
@@ -313,6 +313,14 @@ static TEE_Result ta_load(const TEE_UUID *uuid,
 		goto error_return;
 	}
 
+	if (ta_head->flags & TA_FLAG_MULTI_SESSION &&
```

Comment:
No, we should follow the spec. But I don't want to change any flags, instead whenever we look at for instance `TA_FLAG_INSTANCE_KEEP_ALIVE` we need to check `TA_FLAG_SINGLE_INSTANCE` also because if it's not set then  `TA_FLAG_INSTANCE_KEEP_ALIVE` isn't relevant.